File: ina_2sem/labor_cs/3_lab/1_task/Kwadrat.java
public class Kwadrat extends Czworokat {
    public Kwadrat(double bok, double kat) {
        super(bok, bok, bok, bok, kat);
        if (kat != 90) throw new IllegalArgumentException("Kąt w kwadracie musi wynosić 90°.");
    }

    @Override
    public double obliczPole() {
        return bok1 * bok1;
    }

    @Override
    public double obliczObwod() {
        return 4 * bok1;
    }

    @Override
    public String podajNazwe() {
        return "Kwadrat";
    }
}

File: ina_2sem/labor_cs/3_lab/2_task/Kwadrat.cpp
#include "Kwadrat.h"

Kwadrat::Kwadrat(double bok, double kat) 
    : Czworokat(bok, bok, bok, bok, kat) {
    if (kat != 90) throw std::invalid_argument("Kąt w kwadracie musi wynosić 90°.");
}

double Kwadrat::obliczPole() {
    return bok1 * bok1;
}

double Kwadrat::obliczObwod() {
    return 4 * bok1;
}

std::string Kwadrat::podajNazwe() {
    return "Kwadrat";
}
